rag_experiment_accelerator/checkpoint/checkpoint.py

```diff
@@ -19,15 +19,15 @@ def get_checkpoint():
     return _checkpoint_instance
 
 
-def init_checkpoint(checkpoint_name, config: Config):
+def init_checkpoint(config: Config):
     """
     Initializes the checkpoint instance based on the provided configuration.
     """
     global _checkpoint_instance
-    _checkpoint_instance = _get_checkpoint_base_on_config(checkpoint_name, config)
+    _checkpoint_instance = _get_checkpoint_base_on_config(config)
 
 
-def _get_checkpoint_base_on_config(checkpoint_name, config: Config):
+def _get_checkpoint_base_on_config(config: Config):
```

Comment:
I would move all factory pattern functions to `checkpoint_factory.py` file and leave only the Class there.

Reply:
done